### PR TITLE
SIP-27: Trailing Commas, revised document

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by dwijnand** [comment]
The move back to pending is postponed to ease review as GitHub didn't recognize it as a file rename.

**Comment by dwijnand** [comment]
Review by @xeno-by if possible, please

**Comment by dwijnand** [comment]
The next SIP meeting is next week Tuesday (29th).

If this doesn't get any review, I'm keen to get it merged some time before the meeting to give the committee members a nicely rendered, updated version of the document.

If it were up to me I'd say give it another day and merge it tomorrow if nothing changes, leaving a week for pre-meeting review/study.

Thanks.

**Comment by jvican** [comment]
I agree with the suggestion. Will merge tomorrow if people in the Community don't have anything else to say :smile:.

File: sips/completed/_posts/2016-06-25-trailing-commas.md [hunk 1]
```diff
@@ -95,45 +97,113 @@ Some real-world use-cases where elements of a sequence are typically added, remo
 
 ## Design Decisions
 
-There are a number of different elements of the Scala syntax that are comma separated, but instead of changing them all a subset of the more useful ones was chosen:
+### Multi-line
+
+It is not the intent of introducing trailing commas to promote a code style such as:
```
Comment (xeno-by) [inline sips/completed/_posts/2016-06-25-trailing-commas.md]: Could you briefly explain why?

File: sips/completed/_posts/2016-06-25-trailing-commas.md [hunk 2]
```diff
@@ -95,45 +97,113 @@ Some real-world use-cases where elements of a sequence are typically added, remo
...
+
+With this proposal I would like to present 2 variants:
 
-* tuples
-* argument and parameter groups, including for implicits, for functions, methods and constructors
-* import selectors
+1. The first variant adds trailing comma support to only multi-line `ArgumentExprs`, `Params` and `ClassParams`, which I consider to be the parts of the grammar that would most benefit from trailing commas.
 
-From the spec these are:
+2. The second variant adds trailing comma support to the whole grammar (again, only for multi-line), which means more consistency, but also supporting trailing commas in places that doesn't really need it, such as `ids`.
```
Comment (xeno-by) [inline sips/completed/_posts/2016-06-25-trailing-commas.md]: What do you mean by "doesn't really need it"?

File: sips/completed/_posts/2016-06-25-trailing-commas.md [hunk 3]
```diff
@@ -95,45 +97,113 @@ Some real-world use-cases where elements of a sequence are typically added, remo
...
+  foo: Int,
+  bar: String,
+  baz: Boolean,
+)
+{% endhighlight %}
 
 ## Implementation
 
-The implementation is a simple change to the parser, allowing for a trailing comma, for the groups detailed above, and has been proposed in [scala/scala#5245][].
+The implementation of trailing commas is a matter of changing some of the implementation of Scala's parser. An implementation of this proposal can be found at [scala/scala#5245][].
```
Comment (xeno-by) [inline sips/completed/_posts/2016-06-25-trailing-commas.md]: How about "is limited to changing `Parsers.scala` in the Scala compiler.`?

File: sips/completed/_posts/2016-06-25-trailing-commas.md [hunk 4]
```diff
@@ -3,31 +3,33 @@ layout: sip
 disqus: true
 title: SIP-27 - Trailing Commas
 
-vote-status: accepted 
+vote-status: under review
 vote-text: The following proposal needs to be updated, since only the specialized case version (with new lines) has been accepted. For more information, check the <a href="http://docs.scala-lang.org/sips/minutes/sip-20th-september-minutes.html">minutes</a>.
 ---
 
+// TODO: Move from sips/completed to sips/pending
+
```
Comment (dwijnand) [inline sips/completed/_posts/2016-06-25-trailing-commas.md]: I'm removing the TODO and moving the document in https://github.com/scala/scala.github.com/pull/631, which can be merged after this is merged.

**Comment by jvican** [comment]
@dwijnand Let me know when you're done with all the changes, and I will merge it.

File: sips/completed/_posts/2016-06-25-trailing-commas.md (hunk 1 above)
Comment (dwijnand) [inline sips/completed/_posts/2016-06-25-trailing-commas.md]: done https://github.com/scala/scala.github.com/pull/625/commits/d1ebb53f3257ad27b34d09833af76b2c86872294

**Comment by dwijnand** [comment]
Thank you @xeno-by for the review, see my pushed changes.

@jvican Let's merge either tomorrow or when @xeno-by approves, which ever comes first. Thanks.

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**